src/librbd/image/RemoveRequest.cc

```diff
@@ -621,6 +627,24 @@ void RemoveRequest<I>::handle_remove_v1_image(int r) {
   if (!m_old_format) {
     remove_v2_image();
   }
+
+}
+
+template<typename I>
+void RemoveRequest<I>::remove_image_cache() {
```

Comment:
One possible solution is to run a "rbd ssd cache gc" thread which periodically check the existence of the owner cloned image, and once we find if images are deleted, we just remove those outdated cache.